index.js

```diff
@@ -53,8 +53,9 @@ var initConfig = function(projectConfig) {
   var config = {};
   util._extend(config, require('./config.js'));
   util._extend(config, projectConfig);
-  if (process.env.hasOwnProperty('GCLOUD_TRACE_CONFIG')) {
-    util._extend(config, require(path.resolve(process.env.GCLOUD_TRACE_CONFIG)));
+  if (process.env.hasOwnProperty('GCLOUD_CONFIG')) {
+    util._extend(config,
+      require(path.resolve(process.env.GCLOUD_CONFIG)).trace);
```

Comment:
What exactly do you mean by this? Do you want the contents of the config to be validated to only contain expected properties? If the result of the `require` is a `number` or `function` or `undefined`, this shouldn't break.